src/zenml/integrations/kubeflow/orchestrators/local_deployment_utils.py

```diff
@@ -144,6 +145,37 @@ def kubeflow_pipelines_ready(kubernetes_context: str) -> bool:
         return False
 
 
+def wait_until_kubeflow_pipelines_ready(kubernetes_context: str) -> None:
```

Comment:
I feel like this wait loop logic might be something we encounter quite often. Perhaps something for the `utils` with parameterizable time steps, stop conditions and actions?

Reply:
Makes sense. For the interface, that would get a callable as condition which returns a bool? Should we also add something like a timeout and success/failure log messages?

Reply:
yeah I was also thinking about suggesting a timeout. sounds good

Reply:
@jwwwb I'll merge this as is for the release and create a ticket for refactoring this if that's okay for you, as we're sort of running out of time